### DISCUSSION TITLE
Log file shouldn't be created or written to by default

### DISCUSSION THREAD

**Discussion by cole-h** [helix-editor/helix#10151] [General]
#### Describe your feature request

The log file *should* be truncated, and should only be written to when using one of the verbosity flags. I can see, some day in the (distant?) future, where some user writes a post about how their helix log file was taking up gigabytes of disk space because it's append-only and never cleaned up.

The verbosity flag should take a filename argument that will then be created (if it doesn't exist) or truncated (if it does) and begin writing its logging information to it.

**Comment by archseer** [comment]
Well, we went full circle on this. There was a bug that made us log continually every second when the LSP crashed. Which lead us to [truncate the file each time](https://github.com/helix-editor/helix/issues/735). But then trying to open the logfile (or opening multiple helix instances) would [wipe the file before you could read it](https://github.com/helix-editor/helix/issues/941).

**Comment by cole-h** [comment]
Which is why IMO it should write to stderr by default, or the specified file, and only when run with `-v`.

**Comment by archseer** [comment]
Logging only critical errors is helpful in the case of a crash though. Especially when we get bug reports from stripped builds that don't have a proper backtrace.

**Comment by cole-h** [comment]
In that case, I think a good first step would be to write to stderr by default (and continue logging even without `-v`). If the user wants to put it in a file, it's easy to redirect stderr.

**Comment by kirawi** [comment]
Maybe we could warn the user in Helix if their log is huge. Or maybe that's just lazy, idk.

**Comment by alevinval** [comment]
I looked into logging to `stderr` by default. If we do it, it's going to mess up the editor for most people, as it would force you to always redirect it, to avoid rendering issues as seen in [1].

However, as part of reviewing this ticket, I've opened [a fix for the open-log command](https://github.com/helix-editor/helix/pull/7573)

<details>
<summary> [1] issue with logging by default to stderr </summary>
<img width="845" alt="Screenshot 2023-07-08 at 16 50 00" src="https://github.com/helix-editor/helix/assets/1598610/0d58f2c9-e776-4016-b87b-7c9408437c23">
</details>
